### PR TITLE
#187419065 Fixing productStatus type from string to boolean

### PR DESCRIPTION
**Pull request by YvetteNyibuka**
#### What does this PR do?
Changing product availability status  type from string to boolean

#### Description of Task to be completed?
- Changing product availability status  type from string to boolean
- Making update status an on/off button

#### How should this be manually tested?

Try to access all/single products as buyer and seller and see the results, buyer see only available products

#### What are the relevant pivotal tracker/Trello stories?

#187419065

### CODE REVIEW

**Review by GarrixA — APPROVED** [review]

**Review by cyusasnave — APPROVED** [review]

File: src/middlewares/product.middlewares.ts [hunk 1]
```diff
@@ -45,7 +44,7 @@ export const isProductExist = async (
 	if (!product) {
 		return sendResponse(res, 404, "NOT FOUND", "product is not found");
 	}
-	if (product?.dataValues.productStatus === "Unavailable") {
+	if (product?.dataValues.isAvailable === false) {
```
Comment (leandreAlly) [inline src/middlewares/product.middlewares.ts]: remove false accession and use  (!) operator 
if  (!product?.dataValues.isAvailable ) {
}

File: src/middlewares/product.middlewares.ts [hunk 2]
```diff
@@ -103,7 +74,7 @@ const isProductAvailable = (field: string) => {
 			const product = await database_models.Product.findOne({
 				where: {
 					id: productId,
-					productStatus: "Available",
+					isAvailable: true,
 					expiryDate: {
```
Comment (leandreAlly) [inline src/middlewares/product.middlewares.ts:78]: just use (isAvailable) and remove true.

File: src/mock/static.ts [hunk 3]
```diff
@@ -202,10 +202,10 @@ export const new_update_product = {
 	images: [image_one_path],
 };
 export const new_updated_status = {
-	productStatus: "Available",
+	isAvailable: true,
```
Comment (leandreAlly) [inline src/mock/static.ts]: just use (isAvailable) and remove true.

File: src/mock/static.ts [hunk 4]
```diff
@@ -202,10 +202,10 @@ export const new_update_product = {
...
 };
 export const sameAsOldStatus = {
-	productStatus: "Available",
+	isAvailable: true,
```
Comment (leandreAlly) [inline src/mock/static.ts]: just use (isAvailable) and remove true.

File: src/utils/products.expiration.ts [hunk 5]
```diff
@@ -1,19 +1,18 @@
...
 	const products: ProductAttributes[] = await findAllProducts();
 	const expiredProducts = products.filter((product) => {
 		const expirationDate = new Date(product.expiryDate);
 		const currentDate = new Date();
+
 		return (
 			expirationDate <= currentDate &&
 			product.quantity > 0 &&
-			product.productStatus === "Available"
+			product.isAvailable === true
```
Comment (leandreAlly) [inline src/utils/products.expiration.ts]: remove this "product.isAvailable === true" and have only 
"product.quantity > 0 &&
			product.isAvailable"

File: src/utils/products.expiration.ts [hunk 6]
```diff
@@ -31,20 +30,14 @@ export const checkProductExpiration = async () => {
...
-			to: user.email,
-			subject: `Your products on Hackers have been expired.`,
-			html: update_product_email_template(user, sellerProducts),
-		};
-		sendEmail(options);
 
 		for (const product of sellerProducts) {
 			const condition = { where: { id: product.id } };
-			const productStatus = "Unavailable";
+			const isAvailable = false;
```
Comment (leandreAlly) [inline src/utils/products.expiration.ts:36]: delete this line  const isAvailable = false;

File: src/utils/products.expiration.ts [hunk 7]
```diff
@@ -31,20 +30,14 @@ export const checkProductExpiration = async () => {
...
 			await insert_function<ProductAttributes>(
 				"Product",
 				"update",
-				{ productStatus },
+				{ isAvailable },
```
Comment (leandreAlly) [inline src/utils/products.expiration.ts:40]: replace this with "{ isAvailable }," with "{ !isAvailable }", after deleting the above container variable

File: src/utils/socket.util.ts [hunk 8]
```diff
@@ -17,52 +17,73 @@ export const config = (server: http.Server) => {
...
-			attributes: ["id", "firstName", "lastName", "email", "role"],
-		},
-	];
+	io.use(async (socketInstance, next) => {
+		if (await isLogin(socketInstance)) {
+			next();
+		} else {
+			socketInstance.disconnect(true);
+		}
+	});
```
Comment (leandreAlly) [inline src/utils/socket.util.ts:26]: optional improvement: await isLogin(socketInstance) ? next() : socketInstance.disconnect(true);

File: public/login.html [hunk 9]
```diff
@@ -86,7 +86,7 @@ <h2>Login</h2>
 					email: email,
 					password: password,
 				};
-				fetch("https://hackers-ec-be.onrender.com/api/v1/users/login", {
+				fetch("http://localhost:5500/api/v1/users/login", {
```
Comment (leandreAlly) [inline public/login.html]: put this back this to deployed version!

File: public/login.html [hunk 10]
```diff
@@ -97,8 +97,7 @@ <h2>Login</h2>
 					.then((response) => {
 						if (response.status == "SUCCESS") {
 							localStorage.setItem("loggedUser", JSON.stringify(response.data));
-							window.location.href =
-								"https://hackers-ec-be.onrender.com/api/v1/chats";
+							window.location.href = "http://localhost:5500/api/v1/chats";
```
Comment (leandreAlly) [inline public/login.html]: same here

File: src/__test__/cart.test.ts [hunk 11]
```diff
@@ -229,7 +229,7 @@ describe("CART API TEST", () => {
 
 	it("should return 201 and added to cart successfully", async () => {
 		await database_models.Product.update(
-			{ productStatus: "Available" },
+			{ isAvailable: true },
```
Comment (leandreAlly) [inline src/__test__/cart.test.ts]: also change this

File: src/__test__/order.test.ts [hunk 12]
```diff
@@ -172,7 +172,7 @@ describe("ORDER API TEST", () => {
 
 	it("should return 201 and added to cart successfully", async () => {
 		await database_models.Product.update(
-			{ productStatus: "Available" },
+			{ isAvailable: true },
```
Comment (leandreAlly) [inline src/__test__/order.test.ts]: and here also

File: src/__test__/order.test.ts [hunk 13]
```diff
@@ -221,7 +221,7 @@ describe("ORDER API TEST", () => {
 
 	it("should return 201 and added to cart successfully", async () => {
 		await database_models.Product.update(
-			{ productStatus: "Available", price: 5 },
+			{ isAvailable: true, price: 5 },
```
Comment (leandreAlly) [inline src/__test__/order.test.ts]: and here also change from "{ isAvailable: true,}" to ""{ isAvailable},

File: src/__test__/order.test.ts [hunk 14]
```diff
@@ -242,7 +242,7 @@ describe("ORDER API TEST", () => {
 
 	it("should return 201 and added to cart successfully", async () => {
 		await database_models.Product.update(
-			{ productStatus: "Available", price: 499000 },
+			{ isAvailable: true, price: 499000 },
```
Comment (leandreAlly) [inline src/__test__/order.test.ts]: and here also change from "{ isAvailable: true,}" to ""{ isAvailable},

File: src/__test__/review.test.ts [hunk 15]
```diff
@@ -176,7 +176,7 @@ describe("REVIEW API TEST", () => {
 
 	it("it should  create a review and return 201", async () => {
 		const x = await database_models.Product.update(
-			{ productStatus: "Available" },
+			{ isAvailable: true },
```
Comment (leandreAlly) [inline src/__test__/review.test.ts]: and here also change from "{ isAvailable: true,}" to ""{ isAvailable},

File: src/__test__/sales.test.ts [hunk 16]
```diff
@@ -230,7 +230,7 @@ describe("SALE API TEST", () => {
 
 	it("should return 201 and added to cart successfully", async () => {
 		await database_models.Product.update(
-			{ productStatus: "Available", price: 5 },
+			{ isAvailable: true, price: 5 },
```
Comment (leandreAlly) [inline src/__test__/sales.test.ts]: and here also change from "{ isAvailable: true,}" to ""{ isAvailable},

File: src/__test__/sales.test.ts [hunk 17]
```diff
@@ -251,7 +251,7 @@ describe("SALE API TEST", () => {
 
 	it("should return 201 and added to cart successfully", async () => {
 		await database_models.Product.update(
-			{ productStatus: "Available", price: 499000 },
+			{ isAvailable: true, price: 499000 },
```
Comment (leandreAlly) [inline src/__test__/sales.test.ts]: and here also change from "{ isAvailable: true,}" to ""{ isAvailable},

File: src/controllers/productController.ts [hunk 18]
```diff
@@ -166,7 +165,7 @@ const read_single_product = async (req: Request, res: Response) => {
 		const sellerId = user?.id;
 		const condition_one = { where: { id: product_id, sellerId }, include };
 		const condition_two = {
-			where: { id: product_id, productStatus: "Available" },
+			where: { id: product_id, isAvailable: true },
```
Comment (leandreAlly) [inline src/controllers/productController.ts]: and here also change from "{ isAvailable: true,}" to ""{ isAvailable},

**Review by leandreAlly — CHANGES_REQUESTED** [review]

**Review by Hakizimana-Clement — APPROVED** [review]

**Review by Hakizimana-Clement — APPROVED** [review]

**Review by Hakizimana-Clement — APPROVED** [review]

**Review by Hakizimana-Clement — APPROVED** [review]

**Review by leandreAlly — APPROVED** [review]

**Pull request merged**